homeassistant/auth/providers/legacy_api_password.py

```diff
@@ -28,13 +27,13 @@
 
 
 def _create_repair_and_validate(config: dict[str, Any]) -> dict[str, Any]:
-    async_create_issue(
+    ir.async_create_issue(
         async_get_hass(),
         "auth",
         "deprecated_legacy_api_password",
         breaks_in_ha_version="2024.6.0",
         is_fixable=False,
-        severity=IssueSeverity.WARNING,
+        severity=ir.IssueSeverity.WARNING,
```

Comment:
I like that we have to prepend the functions with `ir.`, but for this class, i always just like to see it as a class. 

(Definitely don't take my word as law, curious what others think)

Reply:
🤷 I don't have a strong opinion on it myself tbh

Reply:
I think for the aliases declared in `[tool.ruff.lint.flake8-import-conventions.extend-aliases]` we should always use the alias prefix.
I think it doesn't make sensor to have two imports - one for the functions and a second one for the classes:
```python
from homeassistant.helpers import issue_registry as ir
from homeassistant.helpers.issue_registry import IssueSeverity
```

And then comes the question about constants...

Reply:
I'll prepare a second PR for [a] components to get a "feel" for it...

Reply:
I opened draft #118218 for components [a]

Reply:
It is quite interesting to find that `er.RegistryEntry` and `ir.IssueSeverity` are both common in the current code base.
`dr.DeviceInfo` much less so currently - that's the biggest change in #118218.

Reply:
I'd agree with @joostlek here. The import aliases are most useful for top level functions which have similar names. All registries have a `async_get`. Just importing the function name and using it would be confusing. Especially in a shortened diff during review.

However, the same doesn't apply to unique class names. There isn't a risk to confuse `DeviceInfo` or `IssueSeverity`. Furthermore, these imports might even be added automatically by IDEs like VSCode with Pylance.

> I think it doesn't make sensor to have two imports - one for the functions and a second one for the classes:
> 
> ```py
> from homeassistant.helpers import issue_registry as ir
> from homeassistant.helpers.issue_registry import IssueSeverity
> ```

Don't believe that's a big deal. There are other instances where we do something like this as well. Mainly when importing constants. The imports are just that: imports. Nothing more.

--
Regarding ruff: Yes, it would be awesome if we could use the build in check, just because it's likely much faster. However, I saw you modified one of our custom pylint plugins in #118215 already. It would be possible to modify it to check just function names instead and recommend the alias import there instead.